### DISCUSSION TITLE
[Bug]: Java heap space my-cluster-kafka

### DISCUSSION THREAD

**Discussion by gustavo84** [strimzi/strimzi-kafka-operator#9603] [Q&A]
### Bug Description

i have this error, the problem is that i want to modify next env variable:

![image](https://github.com/strimzi/strimzi-kafka-operator/assets/16278819/2c38d2fd-0b84-4f02-a787-f122ab3e228d)

![image](https://github.com/strimzi/strimzi-kafka-operator/assets/16278819/963b3833-f866-47c6-8516-f64273a40758)
the steps for replication are:

![image](https://github.com/strimzi/strimzi-kafka-operator/assets/16278819/f0ae2e8f-ac10-4d87-96b8-04da2a96f6a0)
![image](https://github.com/strimzi/strimzi-kafka-operator/assets/16278819/602e5a8f-0503-4f00-91c7-2d305df6d07a)
but i would like to inject/modify the problematic env variable because if i modify to 512 is running, but the problem is that if i modify from the statefulset the environmet variable in a few minuts is changed by the default value:

![image](https://github.com/strimzi/strimzi-kafka-operator/assets/16278819/280c715e-d318-4742-98a2-2ecdb0d7fbad)

### Steps to reproduce

1-go to create kafka
2-create my-cluster-kafa
3-the pod is not running

### Expected behavior

the pod need to run

### Strimzi version

>2.0

### Kubernetes version

1.25

### Installation method

operator 

### Infrastructure

openshift baremetal

**Comment by scholzj** [comment]
If you use AMQ Streams, you should probably contact Red Hat support and not the Strimzi community. In the meantime, as a quick advice:
* Check the docs for how to configure the JVM settings such as Xmx or Xms. It is done through the Kafka custom resource not by modifying some resources managed by the operator.
* You should probably start by setting resource requests and limits for your Kafka cluster as that should help to manage the things such as memory configuration without the need to go to low-level JVM confgiurations.

**Reply by gustavo84** [reply]
its solve with this parameters on kafka kind

jvmOptions:
  "-Xmx": "2g"
  "-Xms": "2g"
